File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/client/MatchLockPacket.java
package pe.nanamochi.banchus.packets.client;

import java.io.IOException;
import java.io.InputStream;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.ClientPacket;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.io.IDataReader;

@Data
@NoArgsConstructor
public class MatchLockPacket implements ClientPacket {
  private int slotId;

  @Override
  public Packets getPacketType() {
    return Packets.OSU_MATCH_LOCK;
  }

  @Override
  public void read(IDataReader reader, InputStream stream) throws IOException {
    this.slotId = reader.readInt32(stream);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/client/MatchLockPacket.kt
package pe.nanamochi.banchus.packets.client

import java.io.InputStream
import pe.nanamochi.banchus.core.BanchoPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataReader

class MatchLockPacket : BanchoPacket(), BanchoPacket.Client {
    override val type = PacketType.OSU_MATCH_LOCK

    var slotId: Int = 0

    override fun read(reader: DataReader, ins: InputStream) {
        this.slotId = reader.readInt32(ins)
    }
}
